railties/lib/rails/secrets.rb

```diff
@@ -30,20 +32,24 @@ def parse(paths, env:)
       end
 
       def generate_key
-        cipher = new_cipher
-        SecureRandom.hex(cipher.key_len)[0, cipher.key_len]
+        new_cipher.random_key
```

Comment:
ISTM this is the outer layer. This class is responsible for receiving the key from the user (3 lines down), so it's going to have to know how to convert it back to raw bytes.

This method is currently a simple pass-through to the cipher doing the real work, so it feels reasonable for this to hexify -- and for the `key` method below to de-hexify. That keeps said work well away from the "real stuff" down in `encrypt`/`decrypt`.